lib/purescript-cst/src/Data/Text/PureScript.hs

```diff
@@ -0,0 +1,23 @@
+-- |
+-- This module contains internal extensions to Data.Text.
+--
+module Data.Text.PureScript (spanUpTo) where
```

Comment:
So, this may be a questionable choice. It's possible to correctly implement `spanUpTo` without using internal and unsafe API from `text`, but only in what I imagine is a somewhat inefficient way—`span` does a Unicode scalar-to-code-point walk on the text, and then `compareLength` does the same, and then if the initial span is too large a `splitAt` does a third walk. I don't know if this makes a real world difference, is the thing. But implementing `spanUpTo` by just tweaking the internal implementation of `span`, as I did here, is very straightforward. I stuffed it in a private module here to limit the consequences of importing these parts of the `text` API elsewhere. As far as premature optimizations go, this one seems pretty benign. But then, they probably all do at first. I can easily rewrite it if this raises flags.